epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -911,12 +913,42 @@ def super_generate_parameters_class_files(self):
             dcoupdecl = [ '      cxtype_sv %s;' % name for name in self.coups_dep ]
             replace_dict['dcoupdecl'] = '\n'.join( dcoupdecl )
             dcoupsetdpar = []
+            # Special handling of G and aS parameters (cudacpp starts from G, while UFO starts from aS)
+            # For simplicity, compute these parameters directly from G, rather than from another such parameter
+            # (e.g. do not compute mdl_sqrt__aS as sqrt of aS, which would require defining aS first)
+            gparameters = { 'aS' : 'G * G / 4. / M_PI',
+                            'mdl_sqrt__aS' : 'G / 2. / constexpr_sqrt( M_PI )' }
+            gparamcoded = set()
             foundG = False
-            for line in self.write_hardcoded_parameters(self.params_dep).split('\n'):
-                if line != '':
-                    dcoupsetdpar.append( '    ' + line.replace('constexpr cxsmpl<double> mdl_G__exp__2','const fptype_sv mdl_G__exp__2').replace('constexpr double', 'const fptype_sv' if foundG else '//const fptype_sv' ) )
-                    if 'constexpr double G =' in line: foundG = True
-            replace_dict['dcoupsetdpar'] = '    ' + '\n'.join( dcoupsetdpar )
+            for pdep in self.params_dep:
+                ###misc.sprint(pdep.type, pdep.name) 
+                line = '    ' + self.write_hardcoded_parameters([pdep]).rstrip('\n')
+                ###misc.sprint(line)
+                if not foundG:
+                    # Comment out the default UFO assignment of mdl_sqrt__aS (from aS) and of G (from mdl_sqrt__aS), but keep them for reference
+                    # (WARNING! This Python CODEGEN code essentially assumes that this refers precisely and only to mdl_sqrt__aS and G)
+                    dcoupsetdpar.append( '    ' + line.replace('constexpr double', '//const fptype_sv') )
+                elif pdep.name == 'mdl_G__exp__2' : # bug fix: fptype, not double nor complex (UFO SUSY and SMEFT even disagree on this?)
+                    # Hardcode a custom assignment (valid for both SUSY and SMEFT) instead of replacing double or complex in 'line'
+                    dcoupsetdpar.append('        const fptype_sv ' + pdep.name + ' = G * G;' )
+                elif pdep.name == 'mdl_G__exp__3' : # bug fix: fptype, not double nor complex (UFO SUSY and SMEFT even disagree on this?)
+                    # Hardcode a custom assignment (valid for both SUSY and SMEFT) instead of replacing double or complex in 'line'
```

Comment:
need to be resolved upstream to always handle it as real.
We can already assume it is always fptype in case

Reply:
PS (after this #632 was already merged). In PR #832 I have now modified these lines, simply adding a comment that mg5amcnlo#89 has been fixed - thanks Olivier